### ISSUE TITLE
Select: Label won't float when using a value with no corresponding option

### ISSUE THREAD

**Issue by robcaldecottvelo** [jamesmfriedman/rmwc#390]
RMWC v4.0.6

This is quite obscure but if you render a `Select` with a `value` but no `options` and then update the `options` later then the label will not float.

```js
class App extends React.Component {
  state = { options: [] };

  render() {
    return (
      <>
        <Select
          outlined
          label="Label"
          options={this.state.options}
          value="Apples"
        />

        <Button
          onClick={() =>
            this.setState({ options: ["Apples", "Oranges"] })
          }
        >
          Update Options
        </Button>
      </>
    );
  }
}
```

There is an example here:

https://codesandbox.io/s/4xv9kkrrm7

We can probably work around this in our code by doing something like:

```js
<Select
  outlined
  label="Label"
  options={this.state.options}
  value={this.state.options.length ? "Apples" : ""}
/>
```

Thoughts?

**Comment by jamesmfriedman** [comment]
You have successfully found every possible way to not float the labe 👀. I’ll take a look.

**Comment by jamesmfriedman** [comment]
Just looking at your example, it's likely to do with an optimization where it doesn't try to recalculate anything if the value doesn't change, but options need to be added to that logic. I'll get a patch release out.

**Comment by jamesmfriedman** [comment]
This is the prescribed fix. First checking options identity to see if there is a diff. Since it is likely that someone is passing in a new object everytime, doing a secondary more expensive check of parsing their options as JSON.

https://github.com/jamesmfriedman/rmwc/commit/441fff996ce10ce2149ef5b6057c7526c93f4dae#diff-b2d52d0b9e6f04df0b23290692d153b2R544

**Comment by jamesmfriedman** [comment]
Fixed in the 5.2.0-alpha.0
